### PR TITLE
Fix overflow in pooling

### PR DESCRIPTION
**Pull request by abergeron**

### CODE REVIEW

**Comment by aam-at** [comment]
[MaxPoolRop](https://github.com/Theano/Theano/blob/master/theano/gpuarray/pool_max_rop.c#L109) should be updated also.

File: theano/gpuarray/tests/test_pool.py [hunk 1]
```diff
@@ -47,6 +47,19 @@ def test_pool_c_interface(self):
             f = theano.function([], ds_op(inp, [2, 2], pad=pad), mode=gpu_mode)
             f()
 
+    def test_pool_big_ws(self):
+        gpu_mode = copy.copy(mode_with_gpu).excluding("cudnn")
```
Comment (nouiz) [inline theano/gpuarray/tests/test_pool.py]: why copy.copy()? Normlally just doing mode_with_gpu.excluding('cudnn') should do the work.

Comment (abergeron) [inline theano/gpuarray/tests/test_pool.py]: I copy pasted the test above (and its mistakes apparently).

Comment (aam-at) [inline theano/gpuarray/tests/test_pool.py]: I also copied (from cuda tests) and I thought that changing `check_py_code` has something to do with `copy`.

**Pull request merged**